framework/include/wifi_manager/wifi_manager.h

```diff
@@ -0,0 +1,163 @@
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef WIFI_MANAGER_H
+#define WIFI_MANAGER_H
+
+typedef enum {
+	// STA mode status
+	AP_DISCONNECTED,
+	AP_CONNECTED,
+
+	// SOFT AP mode status
+	CLIENT_CONNECTED,
+	CLIENT_DISCONNECTED
+} connect_status_e;
+
+/**
+ * @brief <b> wifi result type FAIL, SUCCESS, INVALID ARGS</b>
+ */
+typedef enum {
+	WIFI_MANAGER_FAIL = -1,
+	WIFI_MANAGER_SUCCESS,
+	WIFI_MANAGER_INVALID_ARGS,
+	WIFI_MANAGER_TIMEOUT,
+	WIFI_MANAGER_BUSY,
+} wifi_manager_result_e;
+
+typedef enum {
+	STA_MODE,
+	SOFT_AP_MODE
+} wifi_manager_mode_e;
+
+typedef struct {
+	void (*sta_connected)(void);
+	void (*sta_disconnected)(void);
+	void (*softap_sta_join)(void);
+	void (*softap_sta_leave)(void);
+} wifi_manager_cb_s;
+
+typedef struct {
+	char ip4_address[18];
+	char ssid[32];
+	int rssi;
+	connect_status_e status;
+	wifi_manager_mode_e mode;
+	wifi_manager_cb_s *wmcb;
+} wifi_manager_info_s;
+
+typedef struct {
+	char ssid[32];
+	uint16_t channel;
+	char passphrase[32];
+} wifi_softap_config_s;
+
+/**
+ * @brief <b> wifi authentication type WPA, WPA2, WPS</b>
+ */
+typedef enum {
+	WIFI_MANAGER_AUTH_OPEN,					   /**<  open mode                      */
+	WIFI_MANAGER_AUTH_WEP_SHARED,			   /**<  use shared key (wep key)       */
+	WIFI_MANAGER_AUTH_WPA_PSK,				   /**<  WPA_PSK mode                   */
+	WIFI_MANAGER_AUTH_WPA2_PSK,				   /**<  WPA2_PSK mode                  */
+	WIFI_MANAGER_AUTH_WPA_AND_WPA2_PSK,		   /**<  WPA_PSK and WPA_PSK mixed mode */
+	WIFI_MANAGER_AUTH_UNKNOWN,				   /**<  unknown type                   */
+} wifi_manager_ap_auth_type_e;
+
+/**
+ * @brief wifi encryption type WEP, AES, TKIP
+ */
+typedef enum {
+	WIFI_MANAGER_CRYPTO_NONE,				   /**<  none encryption                */
+	WIFI_MANAGER_CRYPTO_WEP_64,				   /**<  WEP encryption wep-40          */
+	WIFI_MANAGER_CRYPTO_WEP_128,			   /**<  WEP encryption wep-104         */
+	WIFI_MANAGER_CRYPTO_AES,				   /**<  AES encryption                 */
+	WIFI_MANAGER_CRYPTO_TKIP,				   /**<  TKIP encryption                */
+	WIFI_MANAGER_CRYPTO_TKIP_AND_AES,		   /**<  TKIP and AES mixed encryption  */
+	WIFI_MANAGER_CRYPTO_UNKNOWN,			   /**<  unknown encryption             */
+} wifi_manager_ap_crypto_type_e;
+
+/**
+ * @brief wifi ap connect config
+ */
+typedef struct {
+	char ssid[32];							 /**<  Service Set Identification         */
+	unsigned int ssid_length;				 /**<  Service Set Identification Length  */
+	char passphrase[64];					 /**<  ap passphrase(password)            */
+	unsigned int passphrase_length;			 /**<  ap passphrase length               */
+	wifi_manager_ap_auth_type_e ap_auth_type;	  /**<  @ref wifi_utils_ap_auth_type   */
+	wifi_manager_ap_crypto_type_e ap_crypto_type;  /**<  @ref wifi_utils_ap_crypto_type */
+} wifi_manager_ap_config_s;
+
+/**
+* @brief Initialize Wi-Fi Manager including starting Wi-Fi interface.
```

Comment:
Our public APIs should be in group.
If those APIs need new group, you should do "@defgroup " , and include all APIs to that group with "@ingroup"

You can refer to below
/**
 * @defgroup DHCP DHCP
 * @ingroup NETWORK
 * @{
 */

Reply:
Done